cpp/src/gandiva/engine.cc

```diff
@@ -112,9 +113,20 @@ void Engine::InitOnce() {
   }
   ARROW_LOG(INFO) << "Detected CPU Name : " << cpu_name.str();
   ARROW_LOG(INFO) << "Detected CPU Features:" << cpu_attrs_str;
+
+  std::string engine_str = cpu_name.str();
+  engine_str += cpu_attrs_str;
+  engine_str_ = strdup(engine_str.c_str());
```

Comment:
why not just store std::string variable

Reply:
A string variable isn't allowed to be global/static which is why I switched to the char* method.

Reply:
Why would that be so?

Reply:
The linter gave me an error when I tried this not sure why.